From e4cde7c4e8855d4d6fbaaee985f9659263febae2 Mon Sep 17 00:00:00 2001
From: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
Date: Mon, 30 Aug 2021 16:24:49 -0500
Subject: [PATCH 1/1] recipes-phosphor/network: Prevent DHCP release when BMC
 rebooted
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit

Prevents dhcp release message from being sent when BMC is rebooted by
adding the eth0 connection as "critical", this will prevent IP address
change by indicating to DHCP server to preserve the IP for the BMC
client.

Signed-off-by: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
---
 ethernet_interface.cpp | 1 +
 network_config.cpp     | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ethernet_interface.cpp b/ethernet_interface.cpp
index 332fa97..55c4073 100644
--- a/ethernet_interface.cpp
+++ b/ethernet_interface.cpp
@@ -1047,6 +1047,7 @@ void EthernetInterface::writeDHCPSection(std::fstream& stream)
     // Hardcoding the client identifier to mac, to address below issue
     // https://github.com/openbmc/openbmc/issues/1280
     stream << "ClientIdentifier=mac\n";
+    stream << "CriticalConnection=true\n";
     if (manager.getDHCPConf())
     {
         auto value = manager.getDHCPConf()->dnsEnabled() ? "true"s : "false"s;
diff --git a/network_config.cpp b/network_config.cpp
index 94c8eae..767d9bb 100644
--- a/network_config.cpp
+++ b/network_config.cpp
@@ -38,7 +38,8 @@ void writeDHCPDefault(const std::string& filename, const std::string& interface)
                 "IPv6AcceptRA=false\n"

 #endif
-                "[DHCP]\nClientIdentifier=mac\n";
+                "[DHCP]\nClientIdentifier=mac\n"
+                "CriticalConnection=true\n";
     filestream.close();
 }
 } // namespace bmc
--
2.17.1

